aiven/client/connection_info/redis.py

```diff
@@ -0,0 +1,53 @@
+from ._utils import find_component, find_user, format_uri
+from .common import ConnectionInfoError
+
+
+class RedisConnectionInfo:
+    def __init__(self, host, port, username, db, password):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.db = db
```

Comment:
I suggest using `@dataclass` to minimize the amount of boilerplate.

Reply:
I had actually thought of something like that, but I based this code on the other "connection-info" bits, while they're mostly independent, would be nice to keep them with the same interface. Probably best to fix those as well with the same suggestions...